### PR TITLE
Refactoring loading icon display at API request

### PR DESCRIPTION
**Pull request by itutu-tienday**
- Modifications
  - Reorganization of existing code
    - Unified variable names
    - Unification of <Loading /> component coding format
  - Add new loading icon display
    - Workflow screen
      - Initial display
    - Record screen
      - get experiments
      - delete experiment

- Test case
  - Check the loading icon on the following screen
  - standalone mode
    - Workflow screen
      - Initial display
    - Record screen
      - get experiments
      - delete experiment
    - Visualize screen
      - show ROI
  - multiuser mode
    - Add the following to the standalone mode case
    - Workspace screen
    - Account screen
    - AccountManager screen

- Test Result
  - standalone mode

[standalone-loading-icon.webm](https://github.com/arayabrain/barebone-studio/assets/122338895/be8ba254-069b-4a90-bb56-23ad104e9c25)

  - multiuser mode

[multiuser-loading-icon.webm](https://github.com/arayabrain/barebone-studio/assets/122338895/6166bc20-b087-47a7-87ca-5c33bc6f8264)

### CODE REVIEW

**Comment by itutu-tienday** [comment]
@tsuchiyama-araya @milesAraya 
please review this PR.

**Comment by milesAraya** [comment]
I was able to see the loading wheel in Native version. 
Because of the speed of the M3 chip, I needed to add a sleep pause to see the loading wheel. 
Added these lines to [workflow.py](https://github.com/arayabrain/barebone-studio/blob/develop-main/studio/app/common/routers/workflow.py)):
`async def fetch_last_experiment(workspace_id: str):
    last_expt_config = ExptUtils.get_last_experiment(workspace_id)
    import time
    time.sleep(2)
    import logging
    logging.info("Loading icon should be displayed now")`

However, with Docker I was not able to see the loading icon. 
The logging still occurred correctly
> studio-dev-be-1  | 2024-07-08 20:01:28,341 INFO:     [root] fetch_last_experiment():35 - Loading icon should be displayed now

Note, I have had issues with Docker on my PC. The.e is also a chance this is related to using an M3 chip/

**Comment by tsuchiyama-araya** [comment]
こちらをDockerで確認したところ、
Loadingアイコンを確認できませんでした。

行った作業
・feature/refactor-api-loading-iconのブランチに切り替えました
・docker compose up -d でDockerを立ち上げました
・動作確認

File: frontend/src/components/Workspace/Visualize/VisualizeItem.tsx [hunk 1]
```diff
@@ -64,15 +64,15 @@ export const VisualizeItem = memo(function VisualizeItem({
...
+  const loading = useSelector(selectLoading)
   const [roiFilePath, setRoiFilePath] = useState("")
   return (
     <Box
       sx={{ m: 1, display: "flex", flexDirection: "row", position: "relative" }}
     >
-      {loading && roiFilePath.includes("/cell_roi.json") ? (
-        <Loading position={"absolute"} />
-      ) : null}
+      {loading && roiFilePath.includes("_roi.json") && (
```
Comment (tsuchiyama-araya) [inline frontend/src/components/Workspace/Visualize/VisualizeItem.tsx:73]: _roi になった理由を教えていただけませんか？

**Comment by milesAraya** [comment]
The loading icon appeared in native and also in docker in the frontend container.

**Review by milesAraya — APPROVED** [review]

File: frontend/src/components/Workspace/Visualize/VisualizeItem.tsx (hunk 1 above)
Comment (itutu-tienday) [reply frontend/src/components/Workspace/Visualize/VisualizeItem.tsx:73]: @tsuchiyama-araya 
修正前後の動作から確認いただけると思いますが、roiファイルのリクエストは3種類あります（all,cell,none-cell）。
既存コードでは、この内 cell_roi でのみ loading icon が表示されており、UIにやや違和感があったことから、roiリクエストにおけるloading iconの表示条件を統一しています。

**Review by emuemuJP — COMMENTED** [review]
LGTM

**Pull request merged**